coldfront/plugins/qumulo/tasks.py

```diff
@@ -89,6 +90,40 @@ def conditionally_update_storage_allocation_statuses() -> None:
         conditionally_update_storage_allocation_status(allocation)
 
 
+def conditionally_update_billing_cycle_types() -> None:
+    resource = Resource.objects.get(name="Storage2")
+    allocations = Allocation.objects.filter(resources=resource)
+    billing_attribute = AllocationAttributeType.objects.get(name="billing_cycle")
+    prepaid_exp_attribute = AllocationAttributeType.objects.get(
+        name="prepaid_expiration"
+    )
+    billing_sub_q = AllocationAttribute.objects.filter(
+        allocation=OuterRef("pk"), allocation_attribute_type=billing_attribute
+    ).values("value")[:1]
+    prepaid_exp_sub_q = AllocationAttribute.objects.filter(
+        allocation=OuterRef("pk"), allocation_attribute_type=prepaid_exp_attribute
+    ).values("value")[:1]
+    allocations = allocations.annotate(
+        billing_cycle=Subquery(billing_sub_q),
+        prepaid_expiration=Subquery(prepaid_exp_sub_q),
+    )
+    logger.warn(f"Checking billing_cycle in {len(allocations)} qumulo allocations")
+
+    for allocation in allocations:
+        if allocation.billing_cycle == "prepaid":
+            if allocation.prepaid_expiration == datetime.today().strftime(
+                "%Y-%m-%d"
+            ) or allocation.prepaid_expiration < datetime.today().strftime("%Y-%m-%d"):
+                logger.warn(f"Changing {allocation} billing_cycle to monthly")
+                AllocationAttribute.objects.get_or_create(
```

Comment:
@cdeanj2 This is a select or create statement. I recommend appending `.query` to it. Seeing the SQL helps a great deal during development. I have yet to dive deeper into the code; however, the update might be more appropriate if the intention is to update existing records. Good luck!